plugins/package-configuration-providers/dos/src/main/kotlin/DosPackageConfigurationProvider.kt

```
@@ -0,0 +1,164 @@
+
+    override fun getPackageConfigurations(packageId: Identifier, provenance: Provenance): List<PackageConfiguration> {
+        val purl = packageId.toPurl(provenance.toPurlExtras())
+
+        val packageResults = runBlocking { client.getPackageConfiguration(purl) }
+        if (packageResults == null) {
+            logger.info { "Unable to find package configuration for '$purl'." }
+            return emptyList()
+        }
+
+        return if (packageResults.licenseConclusions.isEmpty() && packageResults.pathExclusions.isEmpty()) {
+            emptyList()
+        } else {
+            val packageConfiguration = generatePackageConfiguration(
+                id = packageId,
+                provenance = provenance,
+                packageResults = packageResults
+            )
+
+            logger.info { "Found license conclusions for $purl: ${packageResults.licenseConclusions}" }
+            logger.info { "Found path exclusions for $purl: ${packageResults.pathExclusions}" }
+
+            listOf(packageConfiguration)
+        }
+    }
+}
+
+internal fun generatePackageConfiguration(
+    id: Identifier,
+    provenance: Provenance,
+    packageResults: PackageConfigurationResponseBody
+): PackageConfiguration {
+    val sourceArtifactUrl = if (provenance is ArtifactProvenance) {
+        provenance.sourceArtifact.url
+    } else {
+        null
+    }
+
+    val vcs = if (provenance is RepositoryProvenance) {
+        VcsMatcher(
+            type = provenance.vcsInfo.type,
+            url = provenance.vcsInfo.url,
+            revision = provenance.resolvedRevision
+        )
+    } else {
+        null
+    }
+
+    val licenseFindingCurations = packageResults.licenseConclusions.map { licenseConclusion ->
+        val detected = licenseConclusion.detectedLicenseExpressionSPDX?.takeUnless { it.isEmpty() }
+            ?.let { SpdxExpression.parse(it) }
+
+        check(licenseConclusion.concludedLicenseExpressionSPDX.isNotEmpty())
```

Would it make sense to filter empty ones, and log some info instead of failing hard?

---

This `check` (compared to `require`) is supposed to document in code that this case is never supposed to happen. But if it indeed does, we should indeed fail hard.